app/views/shared/check_answers/_benefits_income.html.erb

```diff
@@ -1,10 +1,18 @@
 <% read_only = false unless local_assigns.key?(:read_only) %>
 
-<% url = if transactions_for.eql?("applicant")
-           providers_legal_aid_application_means_receives_state_benefits_path(@legal_aid_application)
-         else
-           providers_legal_aid_application_partners_receives_state_benefits_path(@legal_aid_application)
-         end %>
+<% if transactions_for.eql?("applicant") %>
+  <% if individual.receives_state_benefits? %>
+    <% url = providers_legal_aid_application_means_add_other_state_benefits_path(@legal_aid_application) %>
+  <% else %>
+    <% url = providers_legal_aid_application_means_receives_state_benefits_path(@legal_aid_application) %>
+  <% end %>
+<% else %>
+  <% if individual.receives_state_benefits? %>
+    <% url = providers_legal_aid_application_partners_add_other_state_benefits_path(@legal_aid_application) %>
+  <% else %>
+    <% url = providers_legal_aid_application_partners_receives_state_benefits_path(@legal_aid_application) %>
+  <% end %>
+<% end %>
```

Comment:
Any suggestions on how to make this more readable are welcomed

Reply:
It could be extracted to a helper to clean up the view?

Reply:
A very good point. I've made the changes now